### PR TITLE
8286797: Guards of constant value false are not permitted

### PR DESCRIPTION
**Pull request by biboudis**
Simple fix that addresses this line from the spec:

> It is a compile-time error if a when expression is a constant expression ([15.29](https://docs.oracle.com/javase/specs/jls/se18/html/jls-15.html#jls-15.29)) with the value false.

Extracts one small utility method `isBooleanWithValue` in `TreeInfo`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286797](https://bugs.openjdk.java.net/browse/JDK-8286797): Guards of constant value false are not permitted

### Reviewers
 * [Vicente Romero](https://openjdk.java.net/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [8814f4fb](https://git.openjdk.java.net/jdk/pull/8745/files/8814f4fb09cb20b8973222f597b4902dbc0dc760)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8745/head:pull/8745` \
`$ git checkout pull/8745`

Update a local copy of the PR: \
`$ git checkout pull/8745` \
`$ git pull https://git.openjdk.java.net/jdk pull/8745/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8745`

View PR using the GUI difftool: \
`$ git pr show -t 8745`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8745.diff">https://git.openjdk.java.net/jdk/pull/8745.diff</a>

</details>

### CODE REVIEW

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties [hunk 1]
```diff
@@ -512,6 +512,9 @@ compiler.err.duplicate.unconditional.pattern=\
 compiler.err.unconditional.pattern.and.default=\
     switch has both an unconditional pattern and a default label
 
+compiler.err.guard.has.constant.expression.false=\
+    this case label has a when constant expression that is false
```
Comment (vicente-romero-oracle) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties:516]: `when` here should be: `''when''`

Comment (biboudis) [reply src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties]: True. I improved the error message a bit.

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties [hunk 2]
```diff
@@ -512,6 +512,9 @@ compiler.err.duplicate.unconditional.pattern=\
 compiler.err.unconditional.pattern.and.default=\
     switch has both an unconditional pattern and a default label
 
+compiler.err.guard.has.constant.expression.false=\
+    this case label has a guard that is a constant expression with value false
```
Comment (vicente-romero-oracle) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties:516]: suggestion:
`... with value ''false''`

**Review by vicente-romero-oracle — APPROVED** [review]
looks good, just a minor suggestion, don't need another iteration

**Pull request closed without merging**